sklearn/utils/estimator_checks.py

```diff
@@ -1122,11 +1149,16 @@ def check_classifiers_train(name, classifier_orig):
                     assert_array_equal(np.argmax(decision, axis=1), y_pred)
 
                 # raises error on malformed input
-                assert_raises(ValueError,
-                              classifier.decision_function, X.T)
+                with assert_raises(ValueError, msg="Malformed inputs"):
+                    classifier.decision_function(X.T)
                 # raises error on malformed input for decision_function
-                assert_raises(ValueError,
-                              classifier.decision_function, X.T)
+                with assert_raises(ValueError, msg="The classifer {name} does "
+                                   "not raise an error when incorrect/malforme"
+                                   "d input for decision_function is passed. N"
```

Comment:
I would really try to make this a bit shorter and maybe make it one sentence. The context of the second sentence is slightly confusing to me. Again, I don't think check_array helps here.

Reply:
" The classifier {} does not raise an error when number of features is inconsistent "
How about this? I am not much familiar with error messages, so sorry for this :(